noauth for a single repo as attribute to a url element

jneuhauser opened the pull request:
Dear devs,

for improved security it should be possible to allow the noauth tag also on a single url element if a unsigned repo is used.

I´ve implemented this for the v8 release, but this patches applies also cleanly on master.

What is your suggestion about this security improvement?

Note: This impl does not break the old noauth impl.

Greets

oldiob commented:
Duplicate of [#220](https://github.com/Linutronix/elbe/issues/220)?

I think that an **attribute** to the _url_ tag would be better than a new element.

~~~xml
<url noauth="true">
...
</url>
~~~

Instead of

~~~xml
<url>
<noauth></noauth>
...
</url>
~~~

Look at the complex type _binary-url_  for an example of adding attribute.

jneuhauser commented:
Sorry for very late reply...

This seems a more clearer way to do it for a single repo url. Thanks for the good input.
Addressed the issues and catched the two aditional cases with "raw-key" elements.

bgermann commented:
I reposted this patch at the elbe-devel mailing list which is used for development. For any comment or newer versions, please use this thread: http://lists.linutronix.de/pipermail/elbe-devel/2019-August/003231.html
